### PR TITLE
Refactor TestValues.cs and Program.cs into more logical classes

### PR DESCRIPTION
**Pull request by stevk**
Really big change that will make additions and improvements easier, but doesn't change the output.
- Each test (Material, Material_Alpha, etc...). has its own class which contains its own values and if statements for building models. All of these are in the 'Tests' folder.
- Log file building has its own class, and the logic I used to make the table strings more human readable are in their own static class.
- The TestValue enums are now in their own file at the namespace level.
- Class for tracking the different parts of an attribute are in their own file now.
- Logic for building combos is in its own static class

In addition
- I've switched from using the word "Parameter" to "Attribute". (I had been using both inconsistently, and parameter seemed more confusing).
- I've replaced some abbreviations in an effort to make variables more readable (mat -> material).
- Some overloaded functions were replaced with optional parameters.

### CODE REVIEW

**Comment by kcoley** [comment]
- [ ] Rename attribute.cs to something else. Maybe property
- [ ] Use switches instead of if else. Especially on the longer ones.
- [ ] Use reflection instead of if statements to run tests
- [ ] Move enums into attributes.cs and testvalues.cs
- [ ] Use reflection to set textures instead of explicitly setting them

**Comment by stevk** [comment]
I only moved some of the tests from if/else statements to using switches, because they were evaluating multiple values in a single check.

File: Source/AssetGeneratorTests/Matrix4x4Tests.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@
 using System.Text;
 using System.Threading.Tasks;
 
-namespace AssetGenerator.Tests
+namespace AssetGenerator.TestNames
```
Comment (bghgary) [inline Source/AssetGeneratorTests/Matrix4x4Tests.cs]: Was this intentional?

File: Source/Attributes.cs [hunk 2]
```diff
@@ -3,29 +3,18 @@
 namespace AssetGenerator
 {
     [AttributeUsage(AttributeTargets.Class)]
-    public class AssetGroupAttribute : Attribute
+    public class TestAttribute : System.Attribute
     {
-        public string Folder { get; private set; }
+        public TestNames TestName { get; private set; }
```
Comment (bghgary) [inline Source/Attributes.cs]: TestNames enum can be singular. TestName property can just be Name.

File: Source/Tests/Material.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections.Generic;
+
+namespace AssetGenerator.Tests
+{
+    [TestAttribute(TestNames.Material),
```
Comment (bghgary) [inline Source/Tests/Material.cs]: Move metadata into TestValues class. TestValues should probably be renamed to Test or TestBase.

**Comment by stevk** [comment]
Feedback actioned.

File: Source/Attributes.cs [hunk 4]
```diff
@@ -5,22 +5,22 @@ namespace AssetGenerator
     [AttributeUsage(AttributeTargets.Class)]
     public class TestAttribute : System.Attribute
     {
-        public TestNames TestName { get; private set; }
+        //public TestName Name { get; private set; }
```
Comment (bghgary) [inline Source/Attributes.cs]: remove commented out code?

File: Source/Tests/Material.cs [hunk 5]
```diff
@@ -2,15 +2,12 @@
 
 namespace AssetGenerator.Tests
 {
-    [TestAttribute(TestNames.Material),
-        ImageAttribute(texture_Emissive),
-        ImageAttribute(texture_Normal),
-        ImageAttribute(texture_OcclusionRoughnessMetallic)]
-    class Material : TestValues
+    [TestAttribute()]
```
Comment (bghgary) [inline Source/Tests/Material.cs]: I think you can omit the `()`

File: Source/Attributes.cs (hunk 4 above)
Comment (stevk) [reply Source/Attributes.cs]: I could leave off the "Attribute" bit too, according to VS. Not sure if I should.

Comment (bghgary) [reply Source/Attributes.cs]: Meh, I usually put it.

**Comment by stevk** [comment]
Fixed the following issues
#134 
#98

**Pull request merged**